abi/cid.go

```diff
@@ -0,0 +1,50 @@
+package abi
+
+import (
+	cid "github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+var (
+	// HashFunction is the default hash function for computing CIDs.
+	//
+	// This is currently Blake2b-256.
+	HashFunction = uint64(mh.BLAKE2B_MIN + 31)
+
+	// When producing a CID for an IPLD block less than or equal to CIDInlineLimit
+	// bytes in length, the identity hash function will be used instead of
+	// HashFunction. This will effectively "inline" the block into the CID, allowing
+	// it to be extracted directly from the CID with no disk/network operations.
+	//
+	// This is currently -1 for "disabled".
+	//
+	// This is exposed for testing. Do not modify unless you know what you're doing.
+	CIDInlineLimit = -1
```

Comment:
@Stebalien I think we should push this down to a member of the cidBuilder, like the codec. I refrained from doing that here.

Reply:
This is here to make it easy to play with. I'd just leave it. The builder is effectively just a singleton.

Reply:
It's singleton until we want to change this, and then it's not.